Makefile

```diff
@@ -52,21 +52,23 @@ clone-tuned:
 	  cd $(TUNED_DIR) && git checkout $(TUNED_COMMIT) && cd .. && \
 	  rm -rf $(TUNED_DIR)/.git)
 
-build: $(BINDATA) pkg/generated build-performance-profile-creator
+build: $(BINDATA) deepcopy pkg/generated build-performance-profile-creator
 	$(GO_BUILD_RECIPE)
 	ln -sf $(PACKAGE_BIN) $(OUT_DIR)/openshift-tuned
 
 $(BINDATA): $(GOBINDATA_BIN) $(ASSETS)
 	$(GOBINDATA_BIN) -mode 420 -modtime 1 -pkg manifests -o $(BINDATA) assets/...
 	gofmt -s -w $(BINDATA)
 
-pkg/generated: $(API_TYPES)
+deepcopy:
 	$(GO) run k8s.io/code-generator/cmd/deepcopy-gen \
 	  --input-dirs $(PACKAGE)/$(API_TYPES_DIR)/tuned/v1,$(PACKAGE)/$(API_TYPES_DIR)/performanceprofile/v1alpha1,$(PACKAGE)/$(API_TYPES_DIR)/performanceprofile/v1,$(PACKAGE)/$(API_TYPES_DIR)/performanceprofile/v2 \
 	  -O $(API_ZZ_GENERATED) \
 	  --go-header-file $(API_GO_HEADER_FILE) \
 	  --bounding-dirs $(PACKAGE)/$(API_TYPES_DIR) \
 	  --output-base tmp
+
+pkg/generated: $(API_TYPES)
```

Comment:
I believe is a bit better if we depend on `deepcopy` here instead of in `build`. IOW:
```
build: $(BINDATA) deepcopy pkg/generated build-performance-profile-creator
...

pkg/generated: $(API_TYPES) deepcopy
```

Reply:
I am unsure because deepcopy files are placed under pkg/apis/../.. directories, so it does not really relate to pkg/generated
but if you insist I can move it.

Reply:
so let's wait for another review to lift the `hold`

Reply:
This change means that every time I make a simple change to NTO code and run `make`, `deepcopy-gen` will run, unnecessarily generating things that do not need to be generated and slowing the compilation down.  Can we not do this please? :)

Reply:
@jmencak Hm but I got another error, that during the change of the API it did not generate the updated deepcopy files. Any ideas on how to have the best of two worlds?